cache/local.go

```diff
@@ -0,0 +1,111 @@
+//   Copyright 2019 MSolution.IO
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package cache
+
+import (
+	"crypto/md5"
+	"database/sql"
+	"fmt"
+	"sort"
+	"strings"
+
+	"github.com/trackit/jsonlog"
+
+	"github.com/trackit/trackit-server/db"
+	"github.com/trackit/trackit-server/models"
+	"github.com/trackit/trackit-server/routes"
+	"github.com/trackit/trackit-server/users"
+)
+
+// formatKey is unique depending on user's AWS' identities (personal + shared accounts)
+// or identities passed in arguments and route's URL
+func formatKey(rdCache *redisCache) {
+	rdCache.key = fmt.Sprintf("%x-%x:", md5.Sum([]byte(rdCache.route)), md5.Sum([]byte(rdCache.args)))
+	for _, val := range rdCache.awsAccount {
+		rdCache.key = fmt.Sprintf("%v%v:", rdCache.key, val)
+	}
+}
+
+func parseRouteFromUrl(url string, rc *redisCache) {
```

Comment:
Try to avoid as much as possible functions that modify values passed to them that don't return anything.

When reading the code from where they are called, it's not obvious what they are modifying.

Reply:
I don't think it's ambiguous since we attribute values to the passed parameter and don't use the rest of the structure.

Reply:
It is ambiguous when you are reading the code that calls this function, _especially_ since it is not commented.

It's clear for you now, but when you are reading code you don't know, you'll have to come to the code of this function and understand it to know what this function do

Reply:
The functions has been commented. I keep in mind your commentary.